pkg/tls/certificatemanagement/render/interface.go

```diff
@@ -0,0 +1,65 @@
+package render
+
+import (
+	"github.com/tigera/operator/pkg/controller/status"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	CASecretName                = "tigera-ca-private"
+	TrustedCertConfigMapName    = "tigera-ca-bundle"
+	TrustedCertConfigMapKeyName = "tigera-ca-bundle.crt"
+	TrustedCertVolumeMountPath  = "/etc/pki/tls/certs/"
+	TrustedCertBundleMountPath  = "/etc/pki/tls/certs/tigera-ca-bundle.crt"
+)
+
+// CertificateManager can sign new certificates and has methods to retrieve existing KeyPairs and Certificates. If a user
+// brings their own secrets, CertificateManager will preserve and return them.
+type CertificateManager interface {
+	// GetKeyPair returns an existing KeyPair. If the KeyPair is not found, nil is returned.
+	GetKeyPair(cli client.Client, secretName, secretNamespace string) (KeyPair, error)
+	// GetOrCreateKeyPair returns a KeyPair. If one exists, some checks are performed. Otherwise, a new KeyPair is created.
+	GetOrCreateKeyPair(cli client.Client, secretName, secretNamespace string, dnsNames []string) (KeyPair, error)
+	// GetCertificate returns a Certificate. If the certificate is not found a k8s.io NotFound error is returned.
+	GetCertificate(cli client.Client, secretName, secretNamespace string) (Certificate, error)
+	// Issued returns true if the provided certificate was signed by CertificateManager and returns false for user provided KeyPairs or Certificate Management.
+	Issued(certificatePem []byte) bool
+	// AddToStatusManager lets the status manager monitor pending CSRs if the certificate management is enabled.
+	AddToStatusManager(manager status.StatusManager, namespace string)
+	KeyPair
```

Comment:
Is there a benefit of having this implement KeyPair instead of having a function, GetCA, which returns a KeyPair for the CA? 

I wouldn't describe this as implementing KeyPair / Certificate, it's actually a bit confusing if you were to read the full function list (it's a key pair / certificate that generates key pairs and certificates).